packages/doenetml-worker-rust/lib-doenetml-core/src/core/props/data_query/data_query.rs

```diff
@@ -81,6 +83,19 @@ pub enum PropSpecifier {
     Matching(Vec<PropProfile>),
 }
 
+/// The source of a  [`DataQuery::PickProp`].
+#[derive(Debug, Clone, PartialEq)]
+pub enum PickPropSource {
+    /// Search for the props on the children of the querying component.
+    Children,
+    /// Search for the prop on ancestors of the querying component.
+    /// The results will be ordered so that the closest ancestor is first in the result.
+    Ancestors,
```

Comment:
Given that you deleted `AncestorRefs` due to its lack of use, you might as well delete `PickPropSource::Ancestors` for the same reason. Unlike descendants, which I know we'll need, I'm not sure if we'll have the need to grab all ancestors.